cmd/src/debug_compose.go

```diff
@@ -114,7 +113,7 @@ func archiveCompose(ctx context.Context, zw *zip.Writer, verbose, noConfigs bool
 	// setup channel for slice of archive function outputs
 	ch := make(chan *archiveFile)
 	g, ctx := errgroup.WithContext(ctx)
-	semaphore := semaphore.NewWeighted(8)
+	semaphore := semaphore.NewWeighted(int64(runtime.GOMAXPROCS(0)))
```

Comment:
This one is interesting, I'm not familiar with `runtime.GOMAXPROCS(0)`, will definitely do some research here 👍

Reply:
It gives you the recommended number of processes/goroutines that can run in parallel (not just concurrently). It's relational to your cpu die count.

Reply:
Hmm, will this potentially reintroduce issues with open file limits? Or reduce the effectiveness of the semaphore in throttling remotely executed requests? In my understanding, this will scale the semaphore's limit in terms of the host machine running `src`'s resources?

Reply:
Number of cpu cores is way under the Linux file handle limit by many orders of magnitude